File: airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/java/io/airbyte/cdk/integrations/destination/s3/writer/DestinationFileWriter.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.cdk.integrations.destination.s3.writer;

import io.airbyte.cdk.integrations.destination.s3.S3Format;

public interface DestinationFileWriter extends DestinationWriter {

  String getFileLocation();

  S3Format getFileFormat();

  String getOutputPath();

}

File: airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/writer/DestinationFileWriter.kt
/*
 * Copyright (c) 2026 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.integrations.destination.s3.writer

import io.airbyte.cdk.integrations.destination.s3.FileUploadFormat

interface DestinationFileWriter : DestinationWriter {
    val fileLocation: String

    val fileFormat: FileUploadFormat?

    val outputPath: String
}
